pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -577,28 +604,136 @@ func (r *SubnetPortReconciler) addressBindingMapFunc(ctx context.Context, obj cl
+		if err != nil {
+			log.Error(err, "Failed to list SubnetPort from cache", "indexValue", spIndexValue)
+			continue
+		}
+		if ab.Spec.InterfaceName == "" && len(spList.Items) > 1 {
+			setAddressBindingStatus(r.Client, ctx, &abList.Items[i], metav1.Now(), multipleInterfaceFoundError, "")
+			continue
+		}
+		found := false
+		for i, sp := range spList.Items {
+			vm, port, err := common.GetVirtualMachineNameForSubnetPort(&spList.Items[i])
+			if err != nil || vm == "" {
+				log.Error(err, "Failed to get VM name from SubnetPort", "namespace", sp.Namespace, "name", sp.Name, "annotations", sp.Annotations)
+				continue
+			}
+			if ab.Spec.InterfaceName == "" || ab.Spec.InterfaceName == port {
+				found = true
+				break
+			}
+		}
+		if !found {
+			setAddressBindingStatus(r.Client, ctx, &abList.Items[i], metav1.Now(), vmOrInterfaceNotFoundError, "")
+		}
+	}
+}
+
+func setAddressBindingStatusBySubnetPort(client client.Client, ctx context.Context, subnetPort *v1alpha1.SubnetPort, subnetPortService *subnetport.SubnetPortService, transitionTime metav1.Time, e error) {
+	ipAddress := ""
 	subnetPortID := subnetPortService.BuildSubnetPortId(&subnetPort.ObjectMeta)
 	nsxSubnetPort := subnetPortService.SubnetPortStore.GetByKey(subnetPortID)
 	if nsxSubnetPort == nil {
 		log.Info("Missing SubnetPort", "id", subnetPort.UID)
-		return
-	}
-	if nsxSubnetPort.ExternalAddressBinding == nil || nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress == nil {
-		return
+		if e == nil {
+			e = vmOrInterfaceNotFoundError
+		}
+	} else if nsxSubnetPort.ExternalAddressBinding != nil && nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress != nil {
+		ipAddress = *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress
 	}
 	ab := subnetPortService.GetAddressBindingBySubnetPort(subnetPort)
 	if ab == nil {
-		log.Info("Missing AddressBinding for SubnetPort", "namespace", subnetPort.Namespace, "name", subnetPort.Name)
+		log.Info("No AddressBinding for SubnetPort", "namespace", subnetPort.Namespace, "name", subnetPort.Name)
 		return
 	}
-	if ab.Status.IPAddress != *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress {
+	setAddressBindingStatus(client, ctx, ab, transitionTime, e, ipAddress)
+}
+
+func setAddressBindingStatus(client client.Client, ctx context.Context, ab *v1alpha1.AddressBinding, transitionTime metav1.Time, e error, ipAddress string) {
+	newConditions := newReadyCondition("AddressBinding", transitionTime, e)
+	isUpdated := false
+	for i := range newConditions {
+		conditionUpdated := false
+		ab.Status.Conditions, conditionUpdated = mergeCondition(ab.Status.Conditions, &newConditions[i])
+		isUpdated = isUpdated || conditionUpdated
+	}
+	if ab.Status.IPAddress != ipAddress {
+		isUpdated = true
+	}
+	if isUpdated {
 		ab = ab.DeepCopy()
-		ab.Status.IPAddress = *nsxSubnetPort.ExternalAddressBinding.ExternalIpAddress
+		ab.Status.IPAddress = ipAddress
 		client.Status().Update(ctx, ab)
```

Comment:
Added error handling for the `client.Status().Update` call?

Reply:
Added err to next line.